gulpfile.js

```diff
@@ -77,100 +78,86 @@ function build(cb) {
+	}
+
+	/**
+	 * @param {string} key
+	 * @param {string} title
+	 */
+	function addLanguageTitle(key, title) {
+		if (!languagesMap[key] && guessTitle(key) !== title) {
+			languagesMap[key] = title;
 		}
-
-		for (const p in data.languages) {
-			if (p !== 'meta') {
-				const title = data.languages[p].displayTitle || data.languages[p].title;
-
-				addLanguageTitle(p, title);
-
-				for (const name in data.languages[p].aliasTitles) {
-					addLanguageTitle(name, data.languages[p].aliasTitles[name]);
-				}
-
-				if (data.languages[p].alias) {
-					if (typeof data.languages[p].alias === 'string') {
-						addLanguageTitle(data.languages[p].alias, title);
-					} else {
-						data.languages[p].alias.forEach(function (alias) {
-							addLanguageTitle(alias, title);
-						});
-					}
+	}
+
+	for (const p in data.languages) {
+		if (p !== 'meta') {
+			const title = data.languages[p].displayTitle || data.languages[p].title;
+			addLanguageTitle(p, title);
+			for (const name in data.languages[p].aliasTitles) {
+				addLanguageTitle(name, data.languages[p].aliasTitles[name]);
+			}
+			if (data.languages[p].alias) {
+				if (typeof data.languages[p].alias === 'string') {
+					addLanguageTitle(data.languages[p].alias, title);
 				}
-
-				if (data.languages[p].require) {
-					dependenciesMap[p] = data.languages[p].require;
+				else {
+					data.languages[p].alias.forEach(function (alias) {
+						addLanguageTitle(alias, title);
+					});
 				}
 			}
+			if (data.languages[p].require) {
+				dependenciesMap[p] = data.languages[p].require;
+			}
 		}
+	}
 
-		const jsonLanguagesMap = JSON.stringify(languagesMap);
-		const jsonDependenciesMap = JSON.stringify(dependenciesMap);
+	const jsonLanguagesMap = JSON.stringify(languagesMap);
+	const jsonDependenciesMap = JSON.stringify(dependenciesMap);
 
-		const tasks = [
-			{ plugin: paths.showLanguagePlugin, map: jsonLanguagesMap },
-			{ plugin: paths.autoloaderPlugin, map: jsonDependenciesMap }
-		];
+	const tasks = [
+		{ plugin: paths.showLanguagePlugin, map: jsonLanguagesMap },
+		{ plugin: paths.autoloaderPlugin, map: jsonDependenciesMap }
+	];
 
-		let cpt = 0;
-		const l = tasks.length;
-		const done = () => {
-			cpt++;
-			if (cpt === l) {
-				cb && cb();
-			}
-		};
-
-		for (const task of tasks) {
-			const stream = src(task.plugin)
-				.pipe(replace(
-					/\/\*languages_placeholder\[\*\/[\s\S]*?\/\*\]\*\//,
-					'/*languages_placeholder[*/' + task.map + '/*]*/'
-				))
-				.pipe(dest(task.plugin.substring(0, task.plugin.lastIndexOf('/'))));
-
-			stream.on('error', done);
-			stream.on('end', done);
-		}
-	});
+	await Promise.all(tasks.map(task => new Promise(resolve => {
+		const stream = src(task.plugin)
+			.pipe(replace(/\/\*languages_placeholder\[\*\/[\s\S]*?\/\*\]\*\//, '/*languages_placeholder[*/' + task.map + '/*]*/'))
+			.pipe(dest(task.plugin.substring(0, task.plugin.lastIndexOf('/'))));
+		stream.on('error', resolve);
```

Comment:
Should we `reject` on `error`?

Reply:
This is the current behaviour.

Reply:
The current behavior is probably a bug, so why not fix it.